src/cart/test/util/cart_logtest.py

```diff
@@ -481,7 +487,126 @@ def _check_pid_from_log_file(self, pid, abort_on_warning,
+                rpc_state = 'SENT'
+
+            elif line.is_callback():
+                rpc = line.descriptor
+                rpc_state = 'COMPLETED'
+                result = line.get_field(-1).rstrip('.')
+                result = C_ERRNOS.get(int(result), result)
+                c_state_names.add(result)
+                opcode = current_opcodes[line.descriptor]
+                try:
+                    c_states[opcode][result] += 1
+                except KeyError:
+
+                    c_states[opcode] = Counter()
+                    c_states[opcode][result] += 1
+            else:
+                continue
+
+            rpc = line.descriptor
+
+            if rpc_state == 'ALLOCATED':
+                current_opcodes[rpc] = opcode
+            else:
+                opcode = current_opcodes[rpc]
+            if rpc_state == 'DEALLOCATED':
+                del current_opcodes[rpc]
+
+            if opcode not in op_state_counters:
+                op_state_counters[opcode] = {'ALLOCATED' :0,
+                                             'DEALLOCATED': 0,
+                                             'SENT':0,
+                                             'COMPLETED':0,
+                                             'SUBMITTED':0}
+            op_state_counters[opcode][rpc_state] += 1
+
+        if not bool(op_state_counters):
+            print('No rpcs in log file')
+            return
+
+        table = []
+        errors = []
+        names = sorted(c_state_names)
+        if names:
+            try:
+                names.remove('-DER_SUCCESS')
+            except ValueError:
+                pass
+            names.insert(0, '-DER_SUCCESS')
+        headers = ['OPCODE',
+                   'ALLOCATED',
+                   'SUBMITTED',
+                   'SENT',
+                   'COMPLETED',
+                   'DEALLOCATED']
+
+        for state in names:
+            headers.append(state)
+        for (op, counts) in sorted(op_state_counters.items()):
+            row = [op,
+                   counts['ALLOCATED'],
+                   counts['SUBMITTED'],
+                   counts['SENT'],
+                   counts['COMPLETED'],
+                   counts['DEALLOCATED']]
+            for state in names:
+                try:
+                    row.append(c_states[op].get(state, ''))
+                except KeyError:
+                    row.append('')
+            table.append(row)
+            if counts['ALLOCATED'] != counts['DEALLOCATED']:
+                errors.append("ERROR: Opcode {}: Alloc'd Total = {}, "
+                              "Dealloc'd Total = {}". \
+                              format(op,
+                                     counts['ALLOCATED'],
+                                     counts['DEALLOCATED']))
+            # Sent can be more than completed because of corpcs but shouldn't
+            # be less
+            if counts['SENT'] > counts['COMPLETED']:
```

Comment:
seems the comment and the check contradict each other

Reply:
Looks like this was a bug before, I'll just remove this check.

https://github.com/daos-stack/iof/blob/master/test/rpctrace_common_methods.py#L246